node-runtime/spec/rest/rest.spec.ts

```diff
@@ -0,0 +1,223 @@
+        },
+        {
+            method: "POST",
+            path: "/add5",
+            result: "1aa",
+            headers: {
+                "x-first": "1",
+            },
+            data: "aa",
+        },
+        {
+            method: "POST",
+            path: "/add5",
+            result: "1aa",
+            headers: {
+                "content-type": "application/json",
+                "x-first": "1",
+            },
+            data: '"aa"',
+        },
+        {
+            method: "POST",
+            path: "/add5",
+            result: '"1aa"',
+            headers: {
+                accept: "application/json",
+                "content-type": "application/json",
+                "x-first": "1",
+            },
+            data: '"aa"',
+        },
+        {
+            method: "GET",
+            path: "/maybe",
+            result: "",
+            statusCode: 404,
+        },
+        {
+            method: "GET",
+            path: "/maybe?bin=4d546864",
+            result: "MThd",
+            resultHeaders: {
+                "content-type": "audio/midi",
+            },
+        },
+        {
+            method: "GET",
+            path: "/maybe?bin=61",
+            result: "a",
+            resultHeaders: {
+                "content-type": "application/octet-stream",
+            },
+        },
+        {
+            method: "GET",
+            path: "/maybe?bin=61",
+            result: '"YQ=="',
+            headers: {
+                accept: "application/json",
+            },
+        },
+        {
+            method: "POST",
+            path: "/maybe",
+            result: "",
+            statusCode: 204,
+        },
+        {
+            method: "POST",
+            path: "/maybe",
+            result: "a",
+            resultHeaders: {
+                "content-type": "application/octet-stream",
+            },
+            data: "61",
+        },
+        {
+            method: "POST",
+            path: "/hex",
+            result: "61",
+            data: "a",
+        },
+        {
+            method: "POST",
+            path: "/obj",
+            result: `{"val":15}`,
+            data: `{"val":15}`,
+            resultHeaders: {
+                "content-type": "application/json",
+            },
+        },
+        {
+            method: "POST",
+            path: "/obj",
+            result: `{"message":"Value is zero ~ spec error","type":"Fatal"}`,
+            data: `{"val":0}`,
+            statusCode: 400,
+            resultHeaders: {
+                "content-type": "application/json",
+            },
+        },
+    ];
+
+    for (const { method, path, result, headers, data, statusCode, resultHeaders } of table) {
+        test(`${method} ${path}` + (headers ? ` with headers ${JSON.stringify(headers)}` : ""), async () => {
+            const response = await axios.request({
+                url: "http://localhost:8001" + path,
+                method: method as any,
```

Comment:
Adding an interface to `table` that allows TypeScript to infer types should be better in this case

Reply:
Done